File: src/main/java/burst/pool/storage/persistent/MinerStore.java
package burst.pool.storage.persistent;

import burst.kit.entity.BurstValue;
import burst.pool.miners.Deadline;

import java.util.List;

public interface MinerStore {
    BurstValue getPendingBalance();
    void setPendingBalance(BurstValue pendingBalance);
    
    double getEstimatedCapacity();
    void setEstimatedCapacity(double estimatedCapacity);
    
    double getShare();
    void setShare(double share);

    BurstValue getMinimumPayout();
    void setMinimumPayout(BurstValue minimumPayout);
    
    String getName();
    void setName(String name);
    
    String getUserAgent();
    void setUserAgent(String userAgent);

    List<Deadline> getDeadlines();
    int getDeadlineCount();
    void removeDeadline(long height);
    Deadline getDeadline(long height);
    void setOrUpdateDeadline(long height, Deadline deadline);

    interface FeeRecipientStore {
        BurstValue getPendingBalance();
        void setPendingBalance(BurstValue pending);
    }
}

File: src/main/kotlin/burst/pool/storage/persistent/MinerStore.kt
package burst.pool.storage.persistent

import burst.kit.entity.BurstValue
import burst.pool.miners.Deadline

interface MinerStore {
    var pendingBalance: BurstValue
    var estimatedCapacity: Double
    var share: Double
    var minimumPayout: BurstValue
    var name: String?
    var userAgent: String?
    val deadlines: List<Deadline>
    val deadlineCount: Int
    fun removeDeadline(height: Long)
    fun getDeadline(height: Long): Deadline?
    fun setOrUpdateDeadline(height: Long, deadline: Deadline)
    interface FeeRecipientStore {
        var pendingBalance: BurstValue
    }
}
